src/LinearAlgebra.java

```diff
@@ -0,0 +1,33 @@
+public class LinearAlgebra {
+
+    // Metodo para transpor uma matriz. A transposição troca as linhas pelas colunas.
+    public static Matrix transpose(Matrix a) {
+        Matrix result = new Matrix(a.getColumns(), a.getRows(), new double[a.getColumns() * a.getRows()]);

```

Comment:
Por estarmos trabalhando com array de elementos da matriz em **float**

```suggestion
        Matrix result = new Matrix(a.getColumns(), a.getRows(), new float[a.getColumns() * a.getRows()]);

```